src/foam/core/Window.js

```diff
@@ -196,12 +207,59 @@ foam.CLASS({
     function cancelAnimationFrame(id) {
       this.window.cancelAnimationFrame(id);
     },
-    function installCSS(text, id, opt_eid) {
-      var eid = foam.u2.Element.NEXT_ID();
-      /* Create a new <style> tag containing the given CSS code. */
-      this.document && this.document.head.insertAdjacentHTML(
-        'beforeend',
-        '<style id="' + opt_eid + '" owner="' + id + '">' + text + '</style>');
+    function expandShortFormMacro(css, m) {
+      /* A short-form macros is of the form %PRIMARY_COLOR%. */
+      var M = m.toUpperCase();
+
+      // NOTE: We add a negative lookahead for */, which is used to close a
+      // comment in CSS. We do this because if we don't, then when a developer
+      // chooses to include a long form CSS macro directly in their CSS such as
+      //
+      //                       /*%EXAMPLE%*/ #abc123
+      //
+      // then we don't want this method to expand the commented portion of that
+      // CSS because it's already in long form. By checking if */ follows the
+      // macro, we can tell if it's already in long form and skip it.
+      return css.replace(
+        new RegExp('%' + M + '%(?!\\*/)', 'g'),
+        '/*%' + M + '%*/ ' + this.theme[m]);
+    },
+
+    function expandLongFormMacro(css, m) {
+      // A long-form macros is of the form "/*%PRIMARY_COLOR%*/ blue".
+      var M = m.toUpperCase();
+
+      return css.replace(
+        new RegExp('/\\*%' + M + '%\\*/[^;]*', 'g'),
+        '/*%' + M + '%*/ ' + this.theme[m]);
+    },
+
+    function installCSS(text, id) {
+      /** CSS preprocessor, works on classes instantiated in subContext. */
+      if ( text ) {
+        var eid = foam.u2.Element.NEXT_ID();
+
+        for ( var i = 0 ; i < this.Theme.MACROS.length ; i++ ) {
+          let m     = this.Theme.MACROS[i];
+          var text2 = this.expandShortFormMacro(this.expandLongFormMacro(text, m), m);
+
+            // If the macro was found, then listen for changes to the property
+            // and update the CSS if it changes.
+            if ( text != text2 ) {
```

Comment:
Note we have a bug here that breaks theme previewing when we use the long form of the CSS. Is this a known bug?

Reply:
I wasn't aware.

Reply:
Yeah when the long form matches the value set by the theme, text1 == tex2 so we don't bind things up.

Reply:
Ahh, that makes sense, thanks. Maybe we need to have a smarter way of binding things. Maybe we can do something like

```JavaScript
if ( /%\w+%/g.test(text) ) { ... }
```

to check if we need to set up the binding.